utils/middleware.ts

```diff
@@ -0,0 +1,37 @@
+import * as t from "io-ts";
+import { PatternString } from "@pagopa/ts-commons/lib/strings";
+import { RequiredQueryParamMiddleware } from "@pagopa/io-functions-commons/dist/src/utils/middlewares/required_query_param";
+import { enumType, withDefault } from "@pagopa/ts-commons/lib/types";
+
+// Tokens are generated by CreateValidationTokenActivity function inside the
+// io-functions-app project (https://github.com/pagopa/io-functions-app)
+// A token is in the following format:
+// [tokenId ULID] + ":" + [validatorHash crypto.randomBytes(12)]
+export const TokenQueryParam = PatternString(
+  "^[A-Za-z0-9]{26}:[A-Fa-f0-9]{24}$"
+);
+export type TokenQueryParam = t.TypeOf<typeof TokenQueryParam>;
+
+export const TokenQueryParamMiddleware = RequiredQueryParamMiddleware(
+  "token",
+  TokenQueryParam
+);
+
+// CONFIRM -> verify token and on success redirect to confirm page
+// VALIDATE -> verify token and on success redirect to result page
+export enum FlowChoiceEnum {
```

Comment:
```suggestion
export enum FlowTypeEnum {
```

Reply:
renamed in https://github.com/pagopa/io-functions-public/pull/72/commits/7d115d231be6393d3b76e2db3ca557f7419a4eab